pallets/xyk/src/lib.rs

```diff
@@ -1366,18 +1327,13 @@ impl<T: Trait> Valuate for Module<T> {
                 Err(_) => return Default::default(),
             };
         let mng_token_reserve = Pools::get((mng_token_id, other_token_id));
-        let liquidity_token_reserve = T::Currency::total_issuance(liquidity_token_id.into());
-        let mng_token_reserve_u256: U256 = mng_token_reserve.saturated_into::<u128>().into();
+        let liquidity_token_reserve: Balance = T::Currency::total_issuance(liquidity_token_id.into()).into();
+        let mng_token_reserve_u256: U256 = mng_token_reserve.into();
         let liquidity_token_amount_u256: U256 =
-            liquidity_token_amount.saturated_into::<u128>().into();
+            liquidity_token_amount.into();
         let liquidity_token_reserve_u256: U256 =
-            liquidity_token_reserve.saturated_into::<u128>().into();
-        let mng_token_amount_u256 =
-            mng_token_reserve_u256 * liquidity_token_amount_u256 / liquidity_token_reserve_u256;
-        let mng_token_amount = mng_token_amount_u256
-            .saturated_into::<u128>()
-            .saturated_into::<Self::Balance>();
-        mng_token_amount
+            liquidity_token_reserve.into();
+        (mng_token_reserve_u256 * liquidity_token_amount_u256 / liquidity_token_reserve_u256).saturated_into()
```

Comment:
why not `multiply_by_rational` here?

Reply:
multiply_by_rational returns an error if the result won't fit into u128.
Here although it seems clear that liquidity_token_amount_u256 / liquidity_token_reserve_u256 will always be less than one, since this function faces outside the pallet it may happen that it is called with inconsistent or de-synced data (Although I currently see no scenario where that is possible with staking, which is the pallet currently using this trait)
So to avoid raising a panic or deal with undefined behaviour ( given that I do not want to change the function signature to use errors, as the part of staking that uses this function does not handle errors gracefully) I do not use multiply_by_rational here.